repos/spack_repo/builtin/packages/hdfview/package.py

```diff
@@ -57,3 +72,14 @@ def setup_build_environment(self, env: EnvironmentModifications) -> None:
         env.set("HDF5LIBS", self.spec["hdf5"].prefix)
         env.set("HDFLIBS", self.spec["hdf"].prefix)
         env.set("ANT_HOME", self.spec["ant"].prefix)
+        env.set("JAVA_HOME ", self.spec["java"].prefix)
+
+    def url_for_version(self, version):
+
+        # the new versions have a complex https://objects.githubusercontent.com url so use the
+        # github release
+        if version >= Version("3.3.0"):
```

Comment:
I wasn't able to curl 3.3.0 from the URL displayed in `spack info` -- is this function right?

Reply:
Great catch thanks. It should be `>` .  I didn't catch the problem as when I tested it (and confirmed just now), spack used its cache papering over the issue

```
==> No binary for hdfview-3.3.0-ycvmnj7277lubbwb4q5bvfqt7qspa6uz found: installing from source
==> Installing hdfview-3.3.0-ycvmnj7277lubbwb4q5bvfqt7qspa6uz [31/31]
==> Fetching https://mirror.spack.io/_source-cache/archive/09/0916161861c21fa8dd354b445b48eff5a53d80a5c0b383e79eb64b7b108e2430.tar.gz
    [100%]  123.83 MB @   17.9 MB/s
==> No patches needed for hdfview

```